### PR TITLE
fix: Service worker category navigation content updating and add subc…

### PR DESCRIPTION
**Pull request by leotoll**
…ategory navigation. Issue: #21

### CODE REVIEW

File: src/hooks/useLazyStore.js [hunk 1]
```diff
@@ -44,7 +43,7 @@ export default function useLazyStore(lazyProps, additionalData = {}) {
         setState(createInitialState)
       }
     }
-  }, [lazyProps.lazy, lazyProps.requestId])
+  }, [lazyProps, lazyProps.requestId])
```
Comment (markbrocato) [inline src/hooks/useLazyStore.js]: Is requestId even needed anymore if we do this?

File: src/link/Link.js [hunk 2]
```diff
@@ -71,7 +71,12 @@ const Link = ({
   } else {
     return (
       <NextLink href={href} prefetch={false} as={as} passHref>
-        {React.cloneElement(children, { ref, onClick: handleClick, ...other })}
+        {React.cloneElement(children, {
+          ref: ref,
+          innerRef: ref,
```
Comment (markbrocato) [inline src/link/Link.js]: What's the purpose of innerRef here?

**Comment by leotoll** [comment]
Those changes are related to tasks: 
[PC-1248](https://moovweb.atlassian.net/browse/PC-1248)
[PC-1234](https://moovweb.atlassian.net/browse/PC-1234)
[PC-1231](https://moovweb.atlassian.net/browse/PC-1231)

**Comment by leotoll** [comment]
Issue #21

File: src/hooks/useLazyStore.js [hunk 3]
```diff
@@ -28,23 +28,37 @@ export default function useLazyStore(lazyProps, additionalData = {}) {
...
+  const updateState = finalState => {
+    if (typeof finalState !== 'function') {
+      stateRef.current = finalState
+
+      return setState(finalState)
+    } else {
+      return setState(state => {
+        stateRef.current = finalState(state)
+
+        return finalState(state)
```
Comment (kevhender) [inline src/hooks/useLazyStore.js]: Are you intentionally calling the function twice? Why not just `return stateRef.current`?

Comment (leotoll) [reply src/hooks/useLazyStore.js]: Good catch!

File: src/link/Link.js (hunk 2 above)
Comment (markbrocato) [reply src/link/Link.js]: Please add comments explaining what innerRef is here.

**Pull request merged**